app/models/assets_model.php

```diff
@@ -56,4 +56,41 @@ function get_assets() {
 
         return $assets_list;
     }
+
+    function get_asset($id) {
+        $assets_list = false;
+
+        $assets = $this->execute(
+            'SELECT * FROM `assets` WHERE `id` = ? ',
+            'i',
+            array(
+                &$id,
+            )
+            );
+
+        if ($assets) {
+            $assets_list = $assets->fetch_all(MYSQLI_ASSOC);
+        }
+
+        return $assets_list;
```

Comment:
You are checking for `empty` in controller and that is the reason it is working.. Anyway, this is just a suggestion. If you strongly believe otherwise, it is fine.